packages/react-components/react-combobox/src/utils/OptionCollection.types.ts

```diff
@@ -35,7 +35,7 @@ export type OptionCollectionState = {
   getOptionById(id: string): OptionValue | undefined;
 
   /** Returns an array of options filtered by a value matching function against the option's value string. */
-  getOptionsMatchingValue(matcher: (value: string) => boolean): OptionValue[];
+  getOptionsMatchingValue(matcher: (value: string | number) => boolean): OptionValue[];
```

Comment:
_nit:_ You could potentially make this `value: OptionValue['value']` so they are always in track in case it changes again in the future.

```suggestion
  getOptionsMatchingValue(matcher: (value: OptionValue['value']) => boolean): OptionValue[];
```